domain/credential/state/state.go

```diff
@@ -882,28 +882,24 @@ func (st *State) ModelsUsingCloudCredential(ctx context.Context, key corecredent
 		return nil, errors.Capture(err)
 	}
 
-	credKey := credentialKey{
-		CredentialName: key.Name,
-		CloudName:      key.Cloud,
-		OwnerName:      key.Owner.Name(),
-	}
-
-	query := `
-SELECT m.* AS &modelNameAndUUID.*
+	stmt, err := st.Prepare(`
+SELECT &modelNameAndUUID.*
 FROM   v_model m
-WHERE  m.cloud_credential_name = $credentialKey.name
-AND    m.cloud_name = $credentialKey.cloud_name
-AND    m.owner_name = $credentialKey.owner_name
-`
-	stmt, err := st.Prepare(query, credKey, modelNameAndUUID{})
+WHERE  m.cloud_credential_uuid = $credentialUUID.uuid
+`, credentialUUID{}, modelNameAndUUID{})
 	if err != nil {
 		return nil, errors.Capture(err)
 	}
 
 	result := make(map[coremodel.UUID]string)
 	var modelNameAndUUIDs []modelNameAndUUID
 	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-		err := tx.Query(ctx, stmt, credKey).GetAll(&modelNameAndUUIDs)
+		uuid, err := st.credentialUUIDForKey(ctx, tx, key)
```

Comment:
Should we look up the UUID from the service and change the CredentialUUIDForKey to take the UUID as arg?

Reply:
Yeah, we already have a bunch of methods which do it this way. The key comes from the caller and doing the lookup here means all the logic is in the same txn. We may revisit, but we'd do it holistically across all the call sites.